rdt-app/src/main/java/io/ona/rdt_app/contract/PatientRegisterActivityContract.java

```diff
@@ -1,12 +1,22 @@
 package io.ona.rdt_app.contract;
 
+import org.json.JSONException;
+
+import io.ona.rdt_app.callback.OnFormSavedCallback;
+
+
 /**
  * Created by Vincent Karuri on 16/07/2019
  */
 public interface PatientRegisterActivityContract {
-    interface View {
+    interface View extends OnFormSavedCallback {
         void openDrawerLayout();
 
         void closeDrawerLayout();
     }
+
+    interface Presenter {
+
+        void saveForm(String jsonForm, OnFormSavedCallback callback) throws JSONException;
```

Comment:
Instead of the presenter throwing the exception and view catching, the presenter on getting `JSONException` should a method on view to display a notification if desired, therefore removing logic from view

Reply:
Makes sense. Done.